circuit_knitting/cutting/cutting_reconstruction.py

```diff
@@ -26,49 +26,70 @@
 
 
 def reconstruct_expectation_values(
-    quasi_dists: Sequence[Sequence[Sequence[tuple[QuasiDistribution, int]]]],
+    results: SamplerResult | dict[str | int, SamplerResult],
     coefficients: Sequence[tuple[float, WeightType]],
     observables: PauliList | dict[str | int, PauliList],
 ) -> list[float]:
     r"""
     Reconstruct an expectation value from the results of the sub-experiments.
 
     Args:
-        quasi_dists: A 3D sequence of length-2 tuples containing the quasi distributions and
-            QPD bit information from each sub-experiment. Its expected shape is
-            (num_unique_samples, num_partitions, num_commuting_observ_groups)
-        coefficients: A sequence of coefficients, such that each coefficient is associated
-            with one unique sample. The length of ``coefficients`` should equal
-            the length of ``quasi_dists``. Each coefficient is a tuple containing the numerical
-            value and the ``WeightType`` denoting how the value was generated.
+        results: The results from running the cutting subexperiments. If the cut circuit
+            was not partitioned between qubits and run separately, the input should be
+            a :class:`~qiskit.primitives.SamplerResult` instance or a dictionary mapping
+            a single partition to the results. If the circuit was partitioned and its
+            pieces run separately, the input should be a dictionary mapping partition labels
+            to the results from each partition's subexperiments.
+
+            The subexperiment results are expected to be ordered in the same way the subexperiments
+            are ordered in the output of :func:`generate_cutting_experiments` -- one result for every
+            sample and observable, as shown below. The Qiskit Sampler primitive will return the results
+            in the same order the experiments are submitted, so users who do not use :func:`generate_cutting_experiments`
+            to generate their experiments should take care to format their subexperiments correctly before submitting them
+            to the sampler primitive.
+
+            :math:`[sample_{0}observable_{0}, \ldots, sample_{0}observable_{N}, sample_{1}observable_{0}, \ldots, sample_{M}observable_{N}]`
+
+        coefficients: The weights associated with each unique subexperiment. Each weight is a tuple
+            containing the scalar value as well as the :class:`WeightType`, which denotes
+            how the value was generated. The contributions from each subexperiment result to the final
+            expectation value will be scaled by this value.
```

Comment:
I might try to avoid the work "weight" here, since the coefficient can be positive _or_ negative.
```suggestion
        coefficients: A sequence containing the coefficient associated with each unique subexperiment. Each element is a tuple
            containing the coefficient (a ``float``) together with its :class:`.WeightType`, which denotes
            how the value was generated. The contribution from each subexperiment will be multiplied by
            its corresponding coefficient, and the resulting terms will be summed to obtain the reconstructed expectation value.
```

Reply:
OK, I've never considered that weight couldn't be negative. I've always just considered it a scalar. We should be consistent though, so this is fine by me